### ISSUE TITLE
Find a suitable CDN

### ISSUE THREAD

**Issue by sebastianhaas** [opensheetmusicdisplay/opensheetmusicdisplay#341]
As discussed in #333 (and #335), it would be beneficial to find a CDN to host OSMD's "binaries", e.g. a minified and a non-minfied version of the JS bundle.

Most CDNs offer to host open source libraries for free, we just need to find one. Let's do that! This issue is to track the progress.

**Comment by sebastianhaas** [comment]
cdnjs.com is off the table for now, they require 200 stars on Github or 800 downloads/month on npm. We don't have either.

https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#b-request-a-new-library

**Comment by sebastianhaas** [comment]
jsDelivr could work, they handle this quite differently. They just fetch files from either npm or Github and then store it on S3 for any future requests:
https://www.jsdelivr.com/features

**Comment by bneumann** [comment]
Jsdelivr already got us covered :
https://www.jsdelivr.com/package/npm/opensheetmusicdisplay

**Comment by sschmidTU** [comment]
this still leaves the problem of #333, that we need to add at least the source map to npm. but i think that's fine, better than putting binaries on github (like vexflow).
Jsdelivr looks good.

**Comment by sebastianhaas** [comment]
Google Hosted won't take us. cdnjs.com would be the best option by far. But let's solve the question with what goes into npm by taking a look at others first, that might open up the possibility to use jsdelivr anyway.

**Comment by sschmidTU** [comment]
Aren't our [Github Releases](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/releases) a CDN? They give static links to the binaries.
I also added the source map to our last release. We could add a non-minified build, but it's pretty big (8MB).

**Comment by sebastianhaas** [comment]
Pretty sure they have rate limits on raw downloads. Also, I'm not sure if linking to raw binaries in the sense of telling others to include a Github raw download in their page headers is in line with GitHub's ToS 🤷

**Comment by ritz078** [comment]
Well you can just use https://unpkg.com . They automatically take updates from GitHub. You just have to add the following in package.json

```js
"unpkg": "link-to-umd-build"
```

**Comment by sebastianhaas** [comment]
But therefore we would need to have a non-minified version within our npm package. Which we don't at the moment.

**Comment by sschmidTU** [comment]
Yes, that only provides the minified release build, not non-minified or source map. We can already get that from cdnjs:
https://cdn.jsdelivr.net/npm/opensheetmusicdisplay@0.5.0/build/opensheetmusicdisplay.min.js

I would rather only add the source map than a non-minified release though.

**Comment by MartinKolarik** [comment]
@sschmidTU just a note that for a source map to work, you also need the original source file. If you don't include it in the npm package, it has to be embedded in the source map file.

**Comment by sschmidTU** [comment]
I know, that's why I said add the source map. We won't make an npm release without the minified build.

**Comment by sschmidTU** [comment]
Our 0.5.0 Github release has minified version + source map for minified version, that's pretty ideal to me. But we weren't sure we wanted to increase our npm package size for the source map.
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/releases/tag/0.5.0

**Comment by sebastianhaas** [comment]
> cdnjs.com is off the table for now, they require 200 stars on Github

Which we will have in the near future :)

**Comment by sschmidTU** [comment]
198 stars already! :)

**Comment by sschmidTU** [comment]
And... 200 stars!

**Comment by sebastianhaas** [comment]
We should definitely consider using cdnjs.com then :) They provide great service.

**Comment by sebastianhaas** [comment]
I created a ticket at cdnjs: https://github.com/cdnjs/cdnjs/issues/13183
